pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -486,15 +486,13 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject, ar
 	// Before starting to process DomainConfig, domainmgr should (in this order):
 	//   1. wait for NIM to publish DNS to learn which ports are used for management
 	//   2. wait for PhysicalIOAdapters (from zedagent) to be processed
-	//   3. wait for NIM to finalize testing of selected DPC
```

Comment:
Did you check the history of why this wait is there?

FWIW I have a vague recollection of either a concern or an actual issue when the initial DPC testing is using interfaces which will be used with PCI passthrough. Maybe this is something which can still happen but since we no longer default to having the fallback to all interfaces the system should be a lot less likely to try other interfaces.

Reply:
I tried to find something in the git history but couldn't really determine the reasons for this wait. If it is to avoid race between nim and domainmgr, in particular "DPC being behind app config" (i.e. NIM testing config which will be replaced by newer DPC anyway and thus potentially breaking already started NIC assignments), then as you pointed out, this should be avoided by using lastresort DPC only when no source of network config is available (and in that case there is no app config to race with).

Reply:
I found https://github.com/lf-edge/eve/commit/dc684a767492878b1852ad18f349a4d4e0ebd764 using git blame. It doesn't clearly point at what failed before the check ...